### PR TITLE
[FE] Cannot lead with a period

### LINKED ISSUE #349: [FE] Cannot lead with a period
When entering the amount, you cannot lead with a period.
e.g. you can't enter `.1` you must enter `0.1`

image

### PR DESCRIPTION
**Pull request by mariano-aguero**
Closes #349 

Peek 05-10-2020 19-26

### CODE REVIEW

File: src/components/common/BigNumberInput/index.tsx [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    value,
+  } = props
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const inputRef = React.useRef<any>(null)
+
+  const [inputValue, setInputvalue] = React.useState('')
+
+  // update current value
+  React.useEffect(() => {
+    const realInputValue = inputValue === '.' || !inputValue ? '0' : inputValue
```
Comment (mariano-aguero) [inline src/components/common/BigNumberInput/index.tsx:33]: Tweek this line, adding a point ('.')

File: src/components/common/BigNumberInput/index.tsx [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    if (!value) {
+      setInputvalue('')
+    } else if (value === '.') {
+      setInputvalue('.')
```
Comment (mariano-aguero) [inline src/components/common/BigNumberInput/index.tsx:37]: Tweek this line, adding a point ('.')

File: src/components/common/BigNumberInput/index.tsx [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    if (!renderInput && autofocus && inputRef) {
+      const node = inputRef.current as HTMLInputElement
+      node.focus()
+    }
+  }, [autofocus, inputRef, renderInput])
+
+  const updateValue = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const { value } = event.currentTarget
+
+    if (value === '' || value === '.') {
```
Comment (mariano-aguero) [inline src/components/common/BigNumberInput/index.tsx:53]: Tweek this line, adding a point ('.')

File: src/components/form/BigNumberInputWrapper/index.tsx [hunk 4]
```diff
@@ -63,7 +63,7 @@ export const BigNumberInputWrapper: React.FC<Props> = (props) => {
 
   const handleChange = (newValue: string) => {
     if (onChange) {
-      if (newValue) {
+      if (newValue && newValue !== '.') {
```
Comment (mariano-aguero) [inline src/components/form/BigNumberInputWrapper/index.tsx:66]: If is a point, must be ZERO

File: src/components/form/BigNumberInputWrapper/index.tsx [hunk 5]
```diff
@@ -38,12 +38,12 @@ const TokenSymbol = styled.span`
...
 
 interface Props {
-  disabled?: boolean | undefined
-  decimals?: number | undefined
+  disabled?: boolean
+  decimals?: number
   onChange?: (n: BigNumber) => void
   tokenSymbol?: string
   value?: BigNumber
-  placeholder?: string | undefined
```
Comment (mariano-aguero) [inline src/components/form/BigNumberInputWrapper/index.tsx:46]: Just some enhancements

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr425--conditionaltokensexplorer.review.gnosisdev.com

**Review by hpmaxi — APPROVED** [review]

**Pull request merged**